core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -535,6 +539,18 @@ private[data] trait EitherTFunctor[F[_], L] extends Functor[EitherT[F, L, ?]] {
   override def map[A, B](fa: EitherT[F, L, A])(f: A => B): EitherT[F, L, B] = fa map f
 }
 
+private[data] trait EitherTApply[F[_], L] extends EitherTFunctor[F, L] with Apply[EitherT[F, L, ?]] {
```

Comment:
This is *very* interesting, and will likely lead to some corrections in cats-mtl as it stands.